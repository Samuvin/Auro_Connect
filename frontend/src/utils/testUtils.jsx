import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Create a custom render function that includes providers
export const renderWithProviders = (ui, options = {}) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  const AllTheProviders = ({ children }) => {
    return (
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

// Mock user data for testing
export const mockUser = {
  _id: '1',
  name: 'John Doe',
  username: 'johndoe',
  email: 'john@example.com',
  profilePicture: '/avatar.png',
  headline: 'Software Developer',
  connections: ['2', '3'],
  location: 'New York, NY'
};

// Mock post data for testing
export const mockPost = {
  _id: '1',
  content: 'This is a test post',
  author: mockUser,
  likes: ['2', '3'],
  comments: [
    {
      _id: '1',
      content: 'Great post!',
      user: mockUser,
      createdAt: new Date().toISOString()
    }
  ],
  image: 'https://example.com/image.jpg',
  createdAt: new Date().toISOString()
}; 