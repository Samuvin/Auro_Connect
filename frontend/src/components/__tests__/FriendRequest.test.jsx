import { screen, fireEvent } from '@testing-library/react';
import FriendRequest from '../FriendRequest';
import { renderWithProviders, mockUser } from '../../utils/testUtils';

describe('FriendRequest', () => {
  const mockRequest = {
    _id: '1',
    sender: mockUser,
    name: mockUser.name, // Component uses request.name for alt attribute
    to: '2',
    createdAt: new Date().toISOString()
  };

  test('renders friend request correctly', () => {
    renderWithProviders(<FriendRequest request={mockRequest} />);
    
    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('Software Developer')).toBeInTheDocument();
  });

  test('shows accept and decline buttons', () => {
    renderWithProviders(<FriendRequest request={mockRequest} />);
    
    expect(screen.getByText(/accept/i)).toBeInTheDocument();
    expect(screen.getByText(/reject/i)).toBeInTheDocument();
  });

  test('displays user profile picture', () => {
    renderWithProviders(<FriendRequest request={mockRequest} />);
    
    const img = screen.getByAltText('John Doe');
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', '/avatar.png');
  });

  test('handles accept button click', () => {
    renderWithProviders(<FriendRequest request={mockRequest} />);
    
    const acceptButton = screen.getByText(/accept/i);
    fireEvent.click(acceptButton);
    
    // Should trigger accept action
    expect(acceptButton).toBeInTheDocument();
  });

  test('handles decline button click', () => {
    renderWithProviders(<FriendRequest request={mockRequest} />);
    
    const rejectButton = screen.getByText(/reject/i);
    fireEvent.click(rejectButton);
    
    // Should trigger reject action
    expect(rejectButton).toBeInTheDocument();
  });

  test('renders with missing profile picture', () => {
    const requestWithoutPicture = {
      ...mockRequest,
      sender: { ...mockUser, profilePicture: null }
    };
    
    renderWithProviders(<FriendRequest request={requestWithoutPicture} />);
    
    const img = screen.getByAltText('John Doe');
    expect(img).toHaveAttribute('src', '/avatar.png');
  });

  // Snapshot Tests
  test('matches snapshot for default friend request', () => {
    const { container } = renderWithProviders(<FriendRequest request={mockRequest} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for friend request with missing profile picture', () => {
    const requestWithoutPicture = {
      ...mockRequest,
      sender: { ...mockUser, profilePicture: null }
    };
    const { container } = renderWithProviders(<FriendRequest request={requestWithoutPicture} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for friend request with different user data', () => {
    const differentUser = {
      ...mockUser,
      name: 'Jane Smith',
      username: 'janesmith',
      headline: 'UX Designer',
      profilePicture: '/jane-avatar.png'
    };
    const requestWithDifferentUser = {
      ...mockRequest,
      sender: differentUser,
      name: differentUser.name
    };
    const { container } = renderWithProviders(<FriendRequest request={requestWithDifferentUser} />);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 