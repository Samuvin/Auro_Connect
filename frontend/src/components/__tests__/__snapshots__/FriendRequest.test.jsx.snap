// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FriendRequest matches snapshot for default friend request 1`] = `
<div
  class="bg-white rounded-lg shadow p-4 flex items-center justify-between transition-all hover:shadow-md"
>
  <div
    class="flex items-center gap-4"
  >
    <a
      href="/profile/johndoe"
    >
      <img
        alt="John Doe"
        class="w-16 h-16 rounded-full object-cover"
        src="/avatar.png"
      />
    </a>
    <div>
      <a
        class="font-semibold text-lg"
        href="/profile/johndoe"
      >
        John Doe
      </a>
      <p
        class="text-gray-600"
      >
        Software Developer
      </p>
    </div>
  </div>
  <div
    class="space-x-2"
  >
    <button
      class="bg-primary text-white px-4 py-2 rounded-md hover:bg-primary-dark transition-colors"
    >
      Accept
    </button>
    <button
      class="bg-gray-200 text-gray-800 px-4 py-2 rounded-md hover:bg-gray-300 transition-colors"
    >
      Reject
    </button>
  </div>
</div>
`;

exports[`FriendRequest matches snapshot for friend request with different user data 1`] = `
<div
  class="bg-white rounded-lg shadow p-4 flex items-center justify-between transition-all hover:shadow-md"
>
  <div
    class="flex items-center gap-4"
  >
    <a
      href="/profile/janesmith"
    >
      <img
        alt="Jane Smith"
        class="w-16 h-16 rounded-full object-cover"
        src="/jane-avatar.png"
      />
    </a>
    <div>
      <a
        class="font-semibold text-lg"
        href="/profile/janesmith"
      >
        Jane Smith
      </a>
      <p
        class="text-gray-600"
      >
        UX Designer
      </p>
    </div>
  </div>
  <div
    class="space-x-2"
  >
    <button
      class="bg-primary text-white px-4 py-2 rounded-md hover:bg-primary-dark transition-colors"
    >
      Accept
    </button>
    <button
      class="bg-gray-200 text-gray-800 px-4 py-2 rounded-md hover:bg-gray-300 transition-colors"
    >
      Reject
    </button>
  </div>
</div>
`;

exports[`FriendRequest matches snapshot for friend request with missing profile picture 1`] = `
<div
  class="bg-white rounded-lg shadow p-4 flex items-center justify-between transition-all hover:shadow-md"
>
  <div
    class="flex items-center gap-4"
  >
    <a
      href="/profile/johndoe"
    >
      <img
        alt="John Doe"
        class="w-16 h-16 rounded-full object-cover"
        src="/avatar.png"
      />
    </a>
    <div>
      <a
        class="font-semibold text-lg"
        href="/profile/johndoe"
      >
        John Doe
      </a>
      <p
        class="text-gray-600"
      >
        Software Developer
      </p>
    </div>
  </div>
  <div
    class="space-x-2"
  >
    <button
      class="bg-primary text-white px-4 py-2 rounded-md hover:bg-primary-dark transition-colors"
    >
      Accept
    </button>
    <button
      class="bg-gray-200 text-gray-800 px-4 py-2 rounded-md hover:bg-gray-300 transition-colors"
    >
      Reject
    </button>
  </div>
</div>
`;
